addons/pos_pinelabs/models/pos_payment_method.py

```diff
@@ -0,0 +1,109 @@
+        })
+        response = pinelabs._call_pinelabs(endpoint='UploadBilledTransaction', payload=body)
+        if not response.get('ResponseCode') and response.get('ResponseMessage') == "APPROVED":
+            return {
+                'responseCode': response.get('ResponseCode'),
+                'status': response.get('ResponseMessage'),
+                'plutusTransactionReferenceID': response.get('PlutusTransactionReferenceID'),
+            }
+        default_error_msg = _('The expected error code for the Pine Labs POS payment upload request was not included in the response.')
+        error = response.get('ResponseMessage') or default_error_msg
+        return {'error': error}
+
+    def pinelabs_fetch_payment_status(self, data):
+        pinelabs = PinelabsPosRequest(self)
+        body = pinelabs._pinelabs_request_body(payment_mode=False)
+        body.update({'PlutusTransactionReferenceID': data.get('plutusTransactionReferenceID')})
+        config = self.env["pos.config"].browse(data.get("config_id"))
+
+        async def get_status():
+            max_retries = 24
+            retry_count = 0
+            try:
+                # Here, we are looping for ResponseCode other than 1001 or retry_count < max_retries
+                while retry_count <= max_retries:
+                    response = pinelabs._call_pinelabs(endpoint='GetCloudBasedTxnStatus', payload=body)
+                    if response.get('ResponseCode') == 1001:
+                        retry_count += 1
+                        await asyncio.sleep(5)  # Non-blocking sleep for 5 seconds
+                    else:
+                        break
+
+                if response.get('ResponseCode') == 0:
+                    pinelabs_latest_response = {
+                        'responseCode': response.get('ResponseCode'),
+                        'status': response.get('ResponseMessage'),
+                        'plutusTransactionReferenceID': response.get('PlutusTransactionReferenceID'),
+                        'data': format_transaction_data(response.get('TransactionData'))
+                    }
+                    config._notify("PINELABS_PAYMENT_RESPONSE", pinelabs_latest_response)
+                else:
+                    default_error_msg = _('The expected error code for the Pine Labs POS payment status request was not included in the response.')
+                    error = response.get('ResponseMessage') or default_error_msg
+                    pinelabs_latest_response = {'error': error}
+                    config._notify("PINELABS_PAYMENT_RESPONSE", pinelabs_latest_response)
+
+            except RequestException:
+                error_msg = _('A request error occurred while checking Pine Labs POS payment status.')
+                pinelabs_latest_response = {'error': error_msg}
+                config._notify("PINELABS_PAYMENT_RESPONSE", pinelabs_latest_response)
+
+        # Initial call to start async the process
+        asyncio.run(get_status())
```

Comment:
i don't think that `asyncio` is part of the odoo requirements. Perhaps you can use `threading.Timer`

Reply:
I initially tried using threading.Timer, but encountered the error `psycopg2.InterfaceError: Cursor already closed` while using `_notifiy`.